toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -217,6 +217,64 @@ func SetupLoopbackDevice(diskFilePath string) (devicePath string, err error) {
 	return
 }
 
+// BlockOnDiskIO waits until all outstanding operations against a disk complete.
+func BlockOnDiskIO(diskDevPath string) (err error) {
+	const (
+		majIdx            = 0
+		minIdx            = 1
+		outstandingOpsIdx = 11
+	)
+
+	logger.Log.Infof("Flushing all IO to disk for %s", diskDevPath)
+	shell.Execute("sync")
```

Comment:
I don't believe sync can return an error here, and even if it did fail, sync is just a request to the OS to perform the flush operation, it will happen in due time on its own and we should block until then anyways.

---
In reply to: [551633956](https://github.com/microsoft/CBL-Mariner/pull/481#discussion_r551633956) [](ancestors = 551633956)

Reply:
I believe we should still check errors here:

`shell.Execute` returns an error in more situations than a non-zero exit code, such as if the main process enters a state where it is no longer allowed to spawn children processes. While rare, we should be defensively handling this.

Reply:
Ok, makes sense.

---
In reply to: [553016117](https://github.com/microsoft/CBL-Mariner/pull/481#discussion_r553016117) [](ancestors = 553016117)